### PR TITLE
Add selection of the imagetype for the Lightbox

### PR DESCRIPTION
**Pull request by MrMusic**
This PR adds the selection of the 'Imagetype' for the Lightbox Gallery.
If the Lightbox Gallery is used in the Category View, only the detail images are currently displayed.
It should be possible to select the original images or detail images.
To do this, the selection of the desired Imagetype is added in the configuration in the Frontend Views -> Category View tab.
config-select-lightbox-imagetype
A new installation is required.

### CODE REVIEW

**Comment by rowi68** [comment]
It works, but when I choose both ( Image link +Title link) for "Link to Lightbox Gallery" so the Title is not clickable anymore, only the picture. 

J 5.2 / PHP 8.2

**Comment by MrMusic** [comment]
>It works, but when I choose both ( Image link +Title link) for "Link to Lightbox Gallery" so the Title is not clickable anymore, only the picture.

That is correct. It is also the case in the current main branch.
The reason is that otherwise all images appear 'twice' in the Lightbox.

**Comment by eumel1602** [comment]
Hallo ! Danke für das Einfügen der Funktion. 
Bei der Einstellung:
- **IMAGE LINK:** Link to eingle Images view
- **Title Link:** Lightbox
- L**ightbox Gallery Image:** Original oder Details (egal welches) 
wird in der angezeigten Lightbox die Vorschaubilder, die unten zu sehen sind, nicht angezeigt bzw sind schwarz.. (Siehe bild)

Screenshot 2024-11-05 125349
**(Kann das jemand bestätigen? getestet auf zwei seiten, auch mit einem anderen PR!!)**
 

> 
> That is correct. It is also the case in the current main branch. The reason is that otherwise all images appear 'twice' in the Lightbox.
Eine Erklärung dazu, sollte in der Config. an dieser Stelle im Kommentar mit auftauchen!

**Comment by reilldesign** [comment]
I have tested this PR ✅ successfully.

File: administrator/com_joomgallery/forms/config.xml [hunk 1]
```diff
@@ -948,6 +948,18 @@
             <option value="defaultview">COM_JOOMGALLERY_CONFIG_CATEGORY_VIEW_LINK_DETAILVIEW</option>
             <option value="lightgallery">COM_JOOMGALLERY_CONFIG_CATEGORY_VIEW_LINK_LIGHTGALLERY</option>
         </field>
+        <field
+            name="jg_category_view_lightbox_image"
+            type="list"
```
[the remark this answers is not in the document]
Comment (MrMusic) [reply administrator/com_joomgallery/forms/config.xml]: Thanks for the suggestion.

File: site/com_joomgallery/tmpl/category/default_cat.php [hunk 2]
```diff
@@ -40,6 +40,7 @@
 $reloaded_images  = $this->params['configs']->get('jg_category_view_number_of_reloaded_images', 3, 'INT');
 $image_link       = $this->params['configs']->get('jg_category_view_image_link', 'defaultview', 'STRING');
 $title_link       = $this->params['configs']->get('jg_category_view_title_link', 'defaultview', 'STRING');
+$lightbox_image   = $this->params['configs']->get('jg_category_view_lightbox_image', 0, 'INT');
```
[the remark this answers is not in the document]
Comment (MrMusic) [reply site/com_joomgallery/tmpl/category/default_cat.php]: Thanks for the suggestion.

File: site/com_joomgallery/tmpl/category/default_cat.php [hunk 3]
```diff
@@ -75,6 +76,14 @@
 $lightbox = false;
 if($image_link == 'lightgallery' || $title_link == 'lightgallery') {
   $lightbox = true;
+  if($lightbox_image == 1)
```
[the remark this answers is not in the document]
Comment (MrMusic) [reply site/com_joomgallery/tmpl/category/default_cat.php]: Thanks for the suggestion.

**Comment by MrMusic** [comment]
>It works, but when I choose both ( Image link +Title link) for "Link to Lightbox Gallery" so the Title is not clickable anymore, only the picture.
That is correct. It is also the case in the current main branch.
The reason is that otherwise all images appear 'twice' in the Lightbox.

@reilldesign Do you know a possibility how to prevent the double loading of the Images in the Lightbox?
Then it would be possible to reactivate the title link.

**Comment by eumel1602** [comment]
Ich installiere den PR neu.
* Ich lege in der Configuration und Imgae processing über den "+" Button einen vierten Imagetyp an. (z.b. riesig) .
Ich ändere  einstellungen im modalfenster (beliebig) 
Schaue ich jetzt in der config unter "Frontend Views " zu Lightbox Gallery Image: Wird mir mein selbst erstellter typ so dargestellt: COM_JOOMGALLERY_RIESIG

* Ich möchte nun diesen selbsterstellten Imagetyp mit dem "-" wieder löschen. Tue ich das und "Speichere" dann die Conif kommt folgende Fehlermeldung: 
`<html><body>
<h1>Es ist ein Fehler aufgetreten.</h1>
                        <blockquote class="blockquote">
                            <span class="badge bg-secondary">1</span>
                            Invalid asset provided for ACL access check                        </blockquote>
                                                    <div>
                                
Call stack
--
# | Function | Location
1 | () | JROOT/administrator/components/com_joomgallery/src/Service/Access/Access.php:480
2 | Joomgallery\Component\Joomgallery\Administrator\Service\Access\Access->prepareAsset() | JROOT/administrator/components/com_joomgallery/src/Service/Access/Access.php:166
3 | Joomgallery\Component\Joomgallery\Administrator\Service\Access\Access->checkACL() | JROOT/administrator/components/com_joomgallery/src/Model/JoomAdminModel.php:676
4 | Joomgallery\Component\Joomgallery\Administrator\Model\JoomAdminModel->canDelete() | JROOT/libraries/src/MVC/Model/AdminModel.php:833
5 | Joomla\CMS\MVC\Model\AdminModel->delete() | JROOT/administrator/components/com_joomgallery/src/Model/ConfigModel.php:424
6 | Joomgallery\Component\Joomgallery\Administrator\Model\ConfigModel->save() | JROOT/libraries/src/MVC/Controller/FormController.php:676
7 | Joomla\CMS\MVC\Controller\FormController->save() | JROOT/libraries/src/MVC/Controller/BaseController.php:693
8 | Joomla\CMS\MVC\Controller\BaseController->execute() | JROOT/administrator/components/com_joomgallery/src/Controller/JoomFormController.php:116
9 | Joomgallery\Component\Joomgallery\Administrator\Controller\JoomFormController->execute() | JROOT/libraries/src/Dispatcher/ComponentDispatcher.php:143
10 | Joomla\CMS\Dispatcher\ComponentDispatcher->dispatch() | JROOT/administrator/components/com_joomgallery/src/Dispatcher/Dispatcher.php:32
11 | Joomgallery\Component\Joomgallery\Administrator\Dispatcher\Dispatcher->dispatch() | JROOT/libraries/src/Component/ComponentHelper.php:361
12 | Joomla\CMS\Component\ComponentHelper::renderComponent() | JROOT/libraries/src/Application/AdministratorApplication.php:143
13 | Joomla\CMS\Application\AdministratorApplication->dispatch() | JROOT/libraries/src/Application/AdministratorApplication.php:186
14 | Joomla\CMS\Application\AdministratorApplication->doExecute() | JROOT/libraries/src/Application/CMSApplication.php:293
15 | Joomla\CMS\Application\CMSApplication->execute() | JROOT/administrator/includes/app.php:61
16 | require_once() | JROOT/administrator/index.php:32

</div>
</body>
</html>`


J4.4.9

**Comment by MrMusic** [comment]
>Ich lege in der Configuration und Imgae processing über den "+" Button einen vierten Imagetyp an. (z.b. riesig) .
Ich ändere einstellungen im modalfenster (beliebig)
Schaue ich jetzt in der config unter "Frontend Views " zu Lightbox Gallery Image: Wird mir mein selbst erstellter typ so dargestellt: 
COM_JOOMGALLERY_RIESIG

Kann ich bestätigen. Das **JgimagetypeField** setzt vor dem selbstgewählten 'Name of Imagetype' noch ein COM_JOOMGALLERY_. Vermutlich damit die drei Standardtypen (original, detail, thumbnail) durch die Sprachdateien übersetzt werden können? Um das Problem zu lösen sehe ich zwei Möglichkeiten:
1. Erstelle im Joomla Backend ein 'Language' override' für COM_JOOMGALLERY_RIESIG und vergib den gewünschten Namen.
2. Wir ersetzen den Teil von `Text::_('COM_JOOMGALLERY_'.\strtoupper($imagetype->typename)` durch `$imagetype->typename`. Dann wird der Name 'unverändert' angezeigt. @Elfangor93 was meinst du dazu?

>Ich möchte nun diesen selbsterstellten Imagetyp mit dem "-" wieder löschen. Tue ich das und "Speichere" dann kommt folgende Fehlermeldung:

Das kann ich auch bestätigen. Der Fehler tritt aber bereits im aktuellen Main-Branch auf und ist offensichtlich bisher noch niemanden aufgefallen.
@Elfangor93 Erstellst du dafür einen separaten Fix?

**Comment by reilldesign** [comment]
> reill

In this case, the code structure must be changed in a later pull request.

**Comment by Elfangor93** [comment]
> Erstellst du dafür einen separaten Fix?

Bitte einen Issue eröffnen.

**Comment by Elfangor93** [comment]
@MrMusic Gibts da noch was zu tun? Können wir den mergen?

**Comment by MrMusic** [comment]
>Wir ersetzen den Teil von Text::_('COM_JOOMGALLERY_'.\strtoupper($imagetype->typename) durch $imagetype->typename. Dann wird der Name 'unverändert' angezeigt. @Elfangor93 was meinst du dazu?

Der Punkt ist noch offen, ob wir aus dem JgimagetypeField die Konstante 'COM_JOOMGALLERY' entfernen oder nicht?

**Comment by Elfangor93** [comment]
> Der Punkt ist noch offen, ob wir aus dem JgimagetypeField die Konstante 'COM_JOOMGALLERY' entfernen oder nicht?

Entweder du überlegst dir was und fügst das in diesem PR hinzu oder wir eröffnen ein issue und kümmern uns später darum.

**Comment by MrMusic** [comment]
>Entweder du überlegst dir was und fügst das in diesem PR hinzu oder wir eröffnen ein issue und kümmern uns später darum.

Ich habe diesen PR nun angepasst.

**Pull request merged**